src/utils/profile.ts

```diff
@@ -21,3 +25,14 @@ export const updateProfileEnteredVenueIds = async (
 
 export const isCompleteProfile = (profile: FirebaseReducer.Profile<User>) =>
   isTruthy(profile.partyName) && isTruthy(profile.pictureUrl);
+
+export const getMappedValues = (mappings: Mappings, source: Object) =>
+  Object.entries(mappings).reduce((acc, [name, path]) => {
+    const mappingValue: string = get(source, path);
+
+    if (!isDefined(mappingValue)) {
+      return acc;
+    }
+
+    return { ...acc, [name]: mappingValue };
+  }, {});
```

Comment:
This seems to show type errors for me:
![image](https://user-images.githubusercontent.com/753891/123575234-a7238300-d814-11eb-92a4-f8ed62d5d5ff.png)

Seems `Object.entries`/`Object.fromEntries` as I suggested in https://github.com/sparkletown/sparkle/pull/1667#discussion_r659067758 isn't typesafe: https://github.com/microsoft/TypeScript/issues/35745

I tried playing around with a few methods.. this was the best I came up with:
```typescript
export const getMappedValues = <
  TObject extends object,
  TKey extends keyof TObject
>(
  mappings: Record<string, TKey>,
  source: TObject
) => mapValues(mappings, (value) => get(source, value, undefined));
```

**Further Reading:**

- https://lodash.com/docs/4.17.15#get
- https://lodash.com/docs/4.17.15#mapValues

Reply:
If we end up using the new method described above, it's super generic, so belongs somewhere in general utils rather than `src/utils/profile.ts` IMO.

(though ultimately it's basically just wrapping those lodash methods anyway, so there may be no point having it extracted as a function and we could just call those methods inline where we need them?

Reply:
(note: my method above may not actually filter the undefined values come to think of it.. unsure if that's needed.. expecting it may not matter to filter them?)

Reply:
Done in https://github.com/sparkletown/sparkle/pull/1667/commits/2d4f2617700b9bb64a4edaab5c88b7bcb223540c + https://github.com/sparkletown/sparkle/pull/1667/commits/c77cd255fb597da0dd2611bc66089e77dbe0e59f